.github/workflows/ci.yml

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) 2023 Files Community
+# Licensed under the MIT License. See the LICENSE.
+
+# Abstract:
+#   This CI is triggered when a commit was added in the main branch directly or
+#   a commit was added in a PR whose base branch is the main branch within this repository.
+
+name: Files CI
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+
+env:
+  APPLICATION_NAME: 'Files'
+  SOLUTION_NAME: 'Files.sln'
+  PRIMARY_ARCHITECTURE: 'x64'
+  SOLUTION_PATH: '${{ github.workspace }}\Files.sln'
+  PACKAGE_PROJECT_DIR: '${{ github.workspace }}\src\Files.App (Package)'
+  PACKAGE_PROJECT_PATH: '${{ github.workspace }}\src\Files.App (Package)\Files.Package.wapproj'
+  INTERACTION_TESTS_PROJECT_DIR: '${{ github.workspace }}\tests\Files.InteractionTests'
+  INTERACTION_TESTS_PROJECT_PATH: '${{ github.workspace }}\tests\Files.InteractionTests\Files.InteractionTests.csproj'
+  WORKING_DIR: ${{ github.workspace }} # Default: 'D:\a\Files\Files'
+  ARTIFACTS_STAGING_DIR:  ${{ github.workspace }}\artifacts
+  APPX_PACKAGE_DIR: ${{ github.workspace }}\artifacts\AppxPackages
+  APPX_SELFSIGNED_CERT_PATH: '${{ github.workspace }}\builds\Files_SelfSigned.pfx'
+
+jobs:
+  build:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Debug]
```

Comment:
Would it be worth having one for release as well?

Reply:
I'm thinking that we can disable building&testing Release config on PRs to keep this fast and enable only on the main branch, like microsoft-ui-xaml team already does?

Reply:
Our existing pipeline is currently set to release. If timing is a concern, we can remove debug and just do release.

Reply:
Debug seems to be faster(build&test on Debug: 15 mins, Release: 20mins) and we can at least run Release only on the main branch. Are you sure?

Reply:
Yes, better to catch issues before merging the PR.

Reply:
> Our existing pipeline is currently set to release. If timing is a concern, we can remove debug and just do release.

Adding or replacing doesn't make any difference because they will be run parallelly and duration will be the same

Reply:
OK!

Reply:
Done